components/camel-as2/camel-as2-api/src/main/java/org/apache/camel/component/as2/api/AS2ClientConnection.java

```diff
@@ -19,42 +19,59 @@
 import java.io.IOException;
 import java.net.Socket;
 import java.net.UnknownHostException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
 
 import org.apache.camel.component.as2.api.io.AS2BHttpClientConnection;
 import org.apache.camel.component.as2.api.protocol.RequestAS2;
 import org.apache.camel.component.as2.api.protocol.RequestMDN;
-import org.apache.http.HttpException;
-import org.apache.http.HttpHost;
-import org.apache.http.HttpRequest;
-import org.apache.http.HttpResponse;
+import org.apache.camel.component.as2.api.util.AS2HttpConnectionFactory;
+import org.apache.http.*;
+import org.apache.http.config.ConnectionConfig;
+import org.apache.http.config.SocketConfig;
+import org.apache.http.conn.*;
+import org.apache.http.conn.HttpConnectionFactory;
+import org.apache.http.conn.routing.HttpRoute;
 import org.apache.http.impl.DefaultBHttpClientConnection;
-import org.apache.http.protocol.HttpCoreContext;
-import org.apache.http.protocol.HttpProcessor;
-import org.apache.http.protocol.HttpProcessorBuilder;
-import org.apache.http.protocol.HttpRequestExecutor;
-import org.apache.http.protocol.RequestConnControl;
-import org.apache.http.protocol.RequestContent;
-import org.apache.http.protocol.RequestDate;
-import org.apache.http.protocol.RequestExpectContinue;
-import org.apache.http.protocol.RequestTargetHost;
-import org.apache.http.protocol.RequestUserAgent;
+import org.apache.http.impl.conn.PoolingHttpClientConnectionManager;
+import org.apache.http.message.BasicHeaderElementIterator;
+import org.apache.http.protocol.*;
 import org.apache.http.util.Args;
 
 public class AS2ClientConnection {
 
     private HttpHost targetHost;
+    private HttpRoute httpRoute;
     private HttpProcessor httpProcessor;
-    private DefaultBHttpClientConnection httpConnection;
     private String as2Version;
     private String userAgent;
     private String clientFqdn;
+    private PoolingHttpClientConnectionManager connectionManager;
+    private ConnectionKeepAliveStrategy connectionKeepAliveStrategy;
+    private IdleConnectionMonitorThread idleConnectionMonitorThread;
 
     public AS2ClientConnection(String as2Version, String userAgent, String clientFqdn, String targetHostName, Integer targetPortNumber) throws UnknownHostException, IOException {
 
         this.as2Version = Args.notNull(as2Version, "as2Version");
         this.userAgent = Args.notNull(userAgent, "userAgent");
         this.clientFqdn = Args.notNull(clientFqdn, "clientFqdn");
         this.targetHost = new HttpHost(Args.notNull(targetHostName, "targetHostName"), Args.notNull(targetPortNumber, "targetPortNumber"));
+        this.connectionManager = new PoolingHttpClientConnectionManager(AS2HttpConnectionFactory.INSTANCE);
+        this.connectionManager.setMaxTotal(3);
+        this.connectionManager.setSocketConfig(targetHost, SocketConfig.custom().setSoTimeout(5000).build());
```

Comment:
It would be nice to add possibility to change parameters like soTimeout, maxTotal and so on, as these can be depends on the environment configuration.